app/models/translation.rb

```diff
@@ -95,18 +95,36 @@ def prevent_destroy_published
   def push_published_to_s3
     return unless is_published
 
-    name_desc_onesky if resource.uses_onesky?
+    phrases = manifest_translated_phrases
+    name_desc_onesky(phrases) if resource.uses_onesky?
+    create_translated_attributes(phrases) if resource.uses_onesky?
 
     Package.new(self).push_to_s3
   end
 
-  def name_desc_onesky
+  def name_desc_onesky(phrases)
     logger.info "Updating translated name and description for translation with id: #{id}"
 
-    p = manifest_translated_phrases
-    self.translated_name = p["name"]
-    self.translated_description = p["description"]
-    self.translated_tagline = p["tagline"]
+    self.translated_name = phrases["name"]
+    self.translated_description = phrases["description"]
+    self.translated_tagline = phrases["tagline"]
+  end
+
+  def create_translated_attributes(phrases)
+    translation_attribute_ids = []
+    resource.translated_attributes.each do |translated_attribute|
+      translation = phrases[translated_attribute.onesky_phrase_id]
+
+      if translation.present?
+        translation_attribute = translation_attributes.where(key: translated_attributes.key).first_or_create
+        translation_attribute.update(value: translation)
+        translation_attribute_ids << translation_attribute.id
+      elsif translated_attribute.required
+        raise Error::TextNotFoundError, "Translated phrase not found: ID: #{translated_attribute.onesky_phrase_id}"
```

Comment:
I'm guessing this doesn't trigger a rollback of other attributes when a missing attribute is encountered? or is this all in a db transaction so a rollback happens automatically?

Reply:
Good question, I believe the attributes already created would remain. I can manually enclose it in a transaction, though. Should I?

Reply:
I would put any updates to the translation object in a single transaction so it rolls back if there is a failure somewhere.

This would include storing the name and description on the translation object, and even storing the manifest file name